### PR TITLE
DNN-10510: Cannot Request Password Reset Link

### PR DESCRIPTION
**Pull request by Mohtshum**
- Unlock the lock out user first before resetting the password.

### CODE REVIEW

File: DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs [hunk 1]
```diff
@@ -1590,8 +1590,12 @@ public override bool ResetAndChangePassword(UserInfo user, string newPassword, s
 
 			//Get AspNet MembershipUser
 			MembershipUser aspnetUser = GetMembershipUser(user);
+            if (aspnetUser.IsLockedOut)
+            {
+                aspnetUser.UnlockUser();
```
Comment (Mohtshum) [inline DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs]: There is a method with AutoUnlock... but it was using time setting.

Comment (kanm2015) [reply DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs]: looks good to me, for resetting pwd no need to go through AutoUnlock expiry chk

**Comment by kanm2015** [comment]
code review done

File: DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs (hunk 1 above)
Comment (galatrash-at-dnn) [reply DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs]: Why we need to auto-unlock the user here? If Admin has locked out any user, then this user can unlock themselves this way which may create a security issue.

Comment (Mohtshum) [reply DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs]: Actually Admin authorize and lockout mostly occur if someone has attempted with wrong password.

**Pull request merged**